recipe/bld.bat

```diff
@@ -32,6 +32,7 @@ set UNIX_SRC_DIR=%SRC_DIR:\=/%
 set PKG_CONFIG_PATH=%UNIX_LIBRARY_PREFIX%/lib/pkgconfig
 
 cmake -LAH -G "Ninja"                                                               ^
+    -DCMAKE_CXX_STANDARD=17                                                         ^
```

Comment:
Seems the error condition at the end should be improved to `if %ERRORLEVEL% neq 0 exit 1` (it appears we're getting an error code that's neither 0 nor 1):

```
-- Configuring incomplete, errors occurred!
CMake Warning:
  Value of OPENCV_BUILD_INFO_STR contained a newline; truncating
```

It might be that, due to the lack of sed-scripting for windows, `CMAKE_CXX_STANDARD=17` now conflicts with those hard-coded values somehow. I suggest that we actually patch those out, and remove the sed scripts. Hopefully we can upstream those changes, but we need that 11->17 on windows most of all, because there the abseil ABI is the most divergent between standard versions.

Reply:
Anyhow the error is not related to the targets that specify `CXX_STANDARD`. For some reason, the code in https://github.com/opencv/opencv/blob/b9ce87e8e28842432e60d9e69b8ad2e66ea647bc/cmake/checks/lapack_check.cpp#L4 does not compile when using C++17 :

~~~
2023-06-06T07:58:24.2890773Z -- found Intel IPP (ICV version): 2020.0.0 [2020.0.0 Gold]
2023-06-06T07:58:24.2892438Z -- at: D:/bld/libopencv_1686037619626/work/build/3rdparty/ippicv/ippicv_win/icv
2023-06-06T07:58:24.2952823Z -- found Intel IPP Integration Wrappers sources: 2020.0.0
2023-06-06T07:58:24.2955309Z -- at: D:/bld/libopencv_1686037619626/work/build/3rdparty/ippicv/ippicv_win/iw
2023-06-06T07:58:24.3566217Z -- LAPACK(Unknown): LAPACK_LIBRARIES: D:/bld/libopencv_1686037619626/_h_env/Library/lib/lapack.lib;D:/bld/libopencv_1686037619626/_h_env/Library/lib/cblas.lib
2023-06-06T07:58:26.0827319Z -- LAPACK(Unknown): Can't build LAPACK check code. This LAPACK version is not supported.
2023-06-06T07:58:26.7267598Z -- Found JNI: C:/Program Files/Microsoft/jdk-11.0.16.101-hotspot/include  found components: AWT JVM 
2023-06-06T07:58:30.7812755Z -- ADE: Downloading v0.1.2a.zip from https://github.com/opencv/ade/archive/v0.1.2a.zip
~~~

Probably we can inspect `CMakeError.log` for more info: https://github.com/opencv/opencv/blob/b9ce87e8e28842432e60d9e69b8ad2e66ea647bc/cmake/OpenCVFindLAPACK.cmake#L118 .

Reply:
Yeah, I had noticed:
```
-- Verifying WITH_LAPACK=1 => 'HAVE_LAPACK'=FALSE
```
but ended up not commenting on it because the list of things here is getting long.

And now looking at the [3rdparty](https://github.com/opencv/opencv/blob/4.x/3rdparty) folder and how stuff gets [pulled](https://github.com/opencv/opencv/blob/4.x/3rdparty/ippicv/ippicv.cmake) from random commits in https://github.com/opencv/opencv_3rdparty, I kinda regret taking a closer look here 😅

Still, I don't see how the snippet you're quoting is actually related to C++17 (or the failure).